src/Draco.FormatterEngine/WhitespaceBehavior.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+
+namespace Draco.Compiler.Internal.Syntax.Formatting;
+
+/// <summary>
+/// Whitespace behavior that will be respected when formatting the file.
+/// </summary>
+[Flags]
+public enum WhitespaceBehavior
+{
+    /// <summary>
+    /// No formatting behavior is set. Unspecified behavior.
+    /// </summary>
+    NoFormatting = 0,
+    /// <summary>
+    /// Add a left whitespace if necessary.
+    /// </summary>
+    SpaceBefore = 1 << 0,
+    /// <summary>
+    /// Add a right whitespace if necessary
+    /// </summary>
+    SpaceAfter = 1 << 1,
+    /// <summary>
+    /// The next token will think of this token as a whitespace.
+    /// </summary>
+    BehaveAsWhiteSpaceForNextToken = 1 << 3,
+    /// <summary>
+    /// The previous token will think of this as a whitespace.
+    /// </summary>
+    BehaveAsWhiteSpaceForPreviousToken = 1 << 4,
+    /// <summary>
+    /// Remove one indentation level.
+    /// </summary>
+    RemoveOneIndentation = 1 << 6,
```

Comment:
Adding one indentation isn't a behavior I suppose?

Reply:
Creating an indentation is simple, you can just create a scope for your line.
Removing an indentation on the other hand is complex, there are tons of edge case, so it's more worth to make a special case for it.